src/swapd/runtime.rs

```diff
@@ -445,6 +468,52 @@ impl Runtime {
         }
     }
 
+    fn continue_deferred_requests(
+        &mut self,
+        endpoints: &mut Endpoints,
+        key: ServiceId,
+        predicate: fn(&PendingRequest) -> bool,
+    ) -> bool {
+        let mut success = true;
+        if let Some(pending_reqs) = self.pending_requests.remove(&key) {
+            let remaining_pending_reqs = pending_reqs
+                .into_iter()
+                .filter_map(|r| {
+                    if predicate(&r) {
+                        if let Ok(_) = match r.bus_id {
+                            ServiceBus::Ctl if r.dest == self.identity => {
+                                self.handle_rpc_ctl(endpoints, r.dest.clone(), r.request.clone())
+                            }
+                            ServiceBus::Msg if r.dest == self.identity => {
+                                self.handle_rpc_msg(endpoints, r.dest.clone(), r.request.clone())
+                            }
+                            _ => endpoints
+                                .send_to(
+                                    r.bus_id.clone(),
+                                    r.source.clone(),
+                                    r.dest.clone(),
+                                    r.request.clone(),
+                                )
+                                .map_err(Into::into),
+                        } {
+                            success = success && true;
```

Comment:
I don't understand this line. Success starts out as true. In this line, it is set to true, if it was true previously, and false if it was already false.